### PR TITLE
CLI tool for pulling and launching Explorer

### PR DESCRIPTION
**Pull request by lbeurerkellner**
Task: https://trello.com/c/DSOUCpuK/161-allow-invariant-explorer-cli-tool-that-fetches-docker-composeyml-from-repo-and-launches-a-local-explorer-instance

This PR allows users to easily pull and launch Explorer on their own machine. For this, the added CLI command `invariant explorer` does the following:

1. It checks the OSS explorer repo for the list of available release versions (v* Git tags)
2. It picks the latest of these release versions (or alternatively `main` if specified via `--version=main`)
3. It downloads/updates the [`docker-compose.yml`](https://github.com/invariantlabs-ai/explorer-public/blob/main/docker-compose.yml) and the [`explorer.local.yml`](https://github.com/invariantlabs-ai/explorer-public/blob/main/configs/explorer.local.yml) file from this repo.
4. It launches a Docker compose stack

The resulting Docker Compose stack will pull in the specified tagged version of the `app-ui` and `app-api` services, as published on `ghcr.io` (GitHub container registry) by CI of the Explorer OSS repo.

This allows us to release Explorer versions via Git tag in the Explorer OSS repo. The only change that this setup still needs is to open source the actual Explorer repo and replace the occurrences of `invariantlabs-a/explorer-public` in this PR. Also, we need to make sure the Explorer OSS repo only pushes versioned and latest images, once CI passes.

### CODE REVIEW

File: testing/invariant/explorer.py [hunk 1]
```diff
@@ -0,0 +1,241 @@
...
+
+def docker_compose_setup(version):
+    """Download the latest Docker Compose setup for the Invariant Explorer."""
+    tf = tempfile.NamedTemporaryFile(
+        delete=False, prefix="docker-compose-", suffix=".yml"
+    )
+
+    with open(tf.name, "w") as f:
+        contents = github_file(
+            "invariantlabs-ai/explorer-public", version, "docker-compose.yml"
```
Comment (lbeurerkellner) [inline testing/invariant/explorer.py:139]: This repository will be pointed to the released OSS repo.

**Comment by Viehzeug** [comment]
The code looks good to me. I assume it should not find containers right now? (Can't fetch manifest right now.)

**Comment by lbeurerkellner** [comment]
It should, but you need to login to GHCR with a GH token (classic) first:

```
echo <TOKEN> | docker login ghcr.io -u <USERNAME> --password-stdin
```

https://docs.github.com/en/packages/working-with-a-github-packages-registry/working-with-the-container-registry#authenticating-with-a-personal-access-token-classic

**Comment by Viehzeug** [comment]
Thanks, that indeed fixed the issue. With this, I could run it, but I found two issues.

1.) I'd name the containers differently than in the explorer repo to allow this setup on a machine as well as the dev setup for the explorer.

2.) I'd keep some flag on whether `invariant explorer` is run for the first time and in this case:
- start all containers
- stop them
- restart
This would avoid the issue with an uninitialized pg instance on the first start. While this is not an issue for the dev instance, it is a a weird issue when accessing it through the CLI.
I think a simple fix would be to check wether `data/database` has any (pg-specific) subfolders. If not, just start the DB briefly to init it.

**Comment by lbeurerkellner** [comment]
I’ve also pushed a version of this now. Maybe you can try. There seems to be an issue with networks sometime, but maybe GH caches are just slow.

**Pull request merged**